pyads/ads.py

```diff
@@ -1107,44 +1107,92 @@ def notification_decorator(func):
-                    value = bytearray(data).split(b"\0", 1)[0].decode("utf-8")
+            return func_wrapper
 
-                elif plc_datatype is not None and issubclass(plc_datatype, Structure):
-                    value = plc_datatype()
-                    fit_size = min(data_size, sizeof(value))
-                    memmove(addressof(value), addressof(data), fit_size)
+        return notification_decorator
 
-                elif plc_datatype is not None and issubclass(plc_datatype, Array):
-                    if data_size == sizeof(plc_datatype):
-                        value = list(plc_datatype.from_buffer_copy(bytes(data)))
-                    else:
-                        # invalid size
-                        value = None
+    def parse_notification(
+        self, notification, plc_datatype, timestamp_as_filetime=False
+    ):
+        # type: (Any, Type, bool) -> (int, int, Any)
+        """Parse a notification.
 
-                elif plc_datatype not in DATATYPE_MAP:
-                    value = bytearray(data)
+        Convert the data of the NotificationHeader into the fitting Python type.
 
-                else:
-                    value = struct.unpack(DATATYPE_MAP[plc_datatype], bytearray(data))[
-                        0
-                    ]
+        :param notification: The notification we recieve from PLC datatype to be
+        converted. This can be any basic PLC datatype or a `ctypes.Structure`.
+        :param plc_datatype: The PLC datatype that needs to be converted. This can
+        be any basic PLC datatype or a `ctypes.Structure`.
+        :param timestamp_as_filetime: Whether the notification timestamp should be returned
+        as `datetime.datetime` (False) or Windows `FILETIME` as originally transmitted
+        via ADS (True). Be aware that the precision of `datetime.datetime` is limited to
+        microseconds, while FILETIME allows for 100 ns. This may be relevant when using
+        task cycle times such as 62.5 µs. Default: False.
 
-                if timestamp_as_filetime:
-                    timestamp = contents.nTimeStamp
-                else:
-                    timestamp = filetime_to_dt(contents.nTimeStamp)
+        :rtype: (int, int, Any)
+        :returns: notification handle, timestamp, value
 
-                return func(contents.hNotification, data_name, timestamp, value)
+        **Usage**:
 
-            return func_wrapper
+        >>> import pyads
+        >>> from ctypes import size_of
+        >>>
+        >>> # Connect to the local TwinCAT PLC
+        >>> plc = pyads.Connection('127.0.0.1.1.1', 851)
+        >>> tag = {"GVL.myvalue": pyads.PLCTYPE_INT}
+        >>>
+        >>> # Create callback function that prints the value
+        >>> def mycallback(notification, data):
+        >>>     data_type = tag[data]
+        >>>     handle, timestamp, value = plc.notification_parser(notification, data_type)
```

Comment:
I suppose this should be plc.parse_notification
```suggestion
        >>>     handle, timestamp, value = plc.parse_notification(notification, data_type)
```

Reply:
Documentation adjusted.